slang-checks/src/test/resources/com/sonarsource/slang/checks/IfConditionalAlwaysTrueOrFalse.slang

```diff
@@ -26,6 +29,8 @@ else if (cond1 || cond2 || true) // Noncompliant
   return 1
 else if (cond && cond2 && !true && cond3) { // Noncompliant
   return 1
+} else if (cond && !(cond2 && (!true && cond3))) { // Noncompliant
```

Comment:
can we test precise location?

Reply:
I'll leave it as it is as the rule highlights the whole condition :)